inference/core/workflows/core_steps/models/third_party/barcode_detection.py

```diff
@@ -72,59 +72,52 @@ def get_manifest(cls) -> Type[WorkflowBlockManifest]:
-            )
-            prediction["predictions"] = detections
-        converted_predictions = attach_prediction_type_info(
-            predictions=predictions,
-            prediction_type="barcode-detection",
-        )
-        converted_predictions = attach_parent_info(
-            images=image, predictions=converted_predictions
-        )
-        return anchor_prediction_detections_in_parent_coordinates(
-            image=image,
-            predictions=converted_predictions,
+        results = []
+        for image in images.iter_nonempty():
+            qr_code_detections = detect_barcodes(image=image)
+            results.append({"predictions": qr_code_detections})
+        return images.align_batch_results(
+            results=results, null_element={"predictions": None}
         )
 
 
-def detect_barcodes(image: np.ndarray) -> List[dict]:
-    barcodes = zxingcpp.read_barcodes(image)
-    predictions = []
+def detect_barcodes(image: WorkflowImageData) -> sv.Detections:
+    barcodes = zxingcpp.read_barcodes(image.numpy_image)
+    xyxy = []
+    confidence = []
+    class_id = []
+    class_name = []
+    extracted_data = []
     for barcode in barcodes:
-        width = barcode.position.top_right.x - barcode.position.top_left.x
-        height = barcode.position.bottom_left.y - barcode.position.top_left.y
-        predictions.append(
-            {
-                "class": "barcode",
-                "class_id": 0,
-                "confidence": 1.0,
-                "x": int(math.floor(barcode.position.top_left.x + width / 2)),
-                "y": int(math.floor(barcode.position.top_left.y + height / 2)),
-                "width": width,
-                "height": height,
-                "detection_id": str(uuid4()),
-                "data": barcode.text,
-            }
-        )
-    return predictions
+        x_min = barcode.position.top_left.x
+        y_min = barcode.position.top_left.y
+        x_max = barcode.position.bottom_right.x
+        y_max = barcode.position.bottom_right.y
+        xyxy.append([x_min, y_min, x_max, y_max])
+        class_id.append(0)
+        class_name.append("barcode")
+        confidence.append(1.0)
+        extracted_data.append(barcode.text)
+    xyxy = np.array(xyxy) if len(xyxy) > 0 else np.empty((0, 4))
+    confidence = np.array(confidence) if len(confidence) > 0 else np.empty(0)
+    class_id = np.array(class_id).astype(int) if len(class_id) > 0 else np.empty(0)
+    class_name = np.array(class_name) if len(class_name) > 0 else np.empty(0)
+    detections = sv.Detections(
+        xyxy=np.array(xyxy),
+        confidence=confidence,
+        class_id=class_id,
+        data={CLASS_NAME_DATA_FIELD: class_name},
+    )
+    detections[DETECTION_ID_KEY] = np.array([uuid4() for _ in range(len(detections))])
+    detections[PREDICTION_TYPE_KEY] = np.array(
+        ["barcode-detection" for _ in range(len(detections))]
```

Comment:
We could avoid loop: `detections[PREDICTION_TYPE_KEY] = np.array(["barcode-detection"]*len(detections))`

Reply:
👍